requests_ftp/ftp.py

```diff
@@ -100,7 +101,46 @@ def build_text_response(request, data, code):
 
 def build_binary_response(request, data, code):
     '''Build a response for data whose encoding is unknown.'''
-    return build_response(request, data, code,  None)
+    return build_response(request, data, code, None)
+
+
+def get_status_code_from_code_response(code):
+    '''
+    The idea is to handle complicated code response (even multi lines).
+    We get the status code in two ways:
+    - extracting the code from the last valid line in the response
+    - getting it from the 3 first digits in the code
+    After a comparison between the two values,
+    we can safely set the code or raise a warning.
+
+    Examples:
+        - get_status_code_from_code_response('200 Welcome') == 200
+
+        - multi_line_code = '226-File successfully transferred\n226 0.000 seconds'
+          get_status_code_from_code_response(multi_line_code) == 226
+
+        - multi_line_with_code_conflicts = '200-File successfully transferred\n226 0.000 seconds'
+          get_status_code_from_code_response(multi_line_with_code_conflicts) == 226
+
+    For more detail see RFC 959, page 36, on multi-line responses:
+        https://www.ietf.org/rfc/rfc959.txt
+
+        "Thus the format for multi-line replies is that the first line
+         will begin with the exact required reply code, followed
+         immediately by a Hyphen, "-" (also known as Minus), followed by
+         text.  The last line will begin with the same code, followed
+         immediately by Space <SP>, optionally some text, and the Telnet
+         end-of-line code."
+    '''
+    last_valid_line_from_code = [line for line in code.split('\n') if line][-1]
+    status_code_from_last_line = int(last_valid_line_from_code.split()[0])
+    status_code_from_first_digits = int(code[:3])
+    if status_code_from_last_line != status_code_from_first_digits:
+        logging.warning(
+            'FTP response status code seems to be inconsistent.\n'
+            'Code received: {0}, extracted: {1} and {2}'.format(
+                code, status_code_from_last_line, status_code_from_first_digits))
```

Comment:
Additionally, when logging it's better to use the logger's string interpolation logic rather than calling `format`. That's because the logger will skip the interpolation if the logs at this level are being suppressed.